drainer/k8s_utils.py

```diff
@@ -28,19 +30,37 @@ def cordon_node(api, node_name):
 
 def remove_all_pods(api, node_name, poll=5):
     """Removes all Kubernetes pods from the specified node."""
-    field_selector = 'spec.nodeName=' + node_name
-    pods = api.list_pod_for_all_namespaces(watch=False, field_selector=field_selector)
+    pods = get_evictable_pods(api, node_name)
+
+    logger.debug('Number of pods to delete: ' + str(len(pods)))
+
+    evict_until_completed(api, pods, poll)
+    wait_until_empty(api, node_name, poll)
 
-    logger.debug('Number of pods to delete: ' + str(len(pods.items)))
 
-    try_until_completed(api, evict_pods, pods, poll)
-    try_until_completed(api, get_pending, pods, poll)
+def pod_is_evictable(pod):
+    if pod.metadata.annotations is not None and pod.metadata.annotations.get(MIRROR_POD_ANNOTATION_KEY):
+        logger.info("Skipping mirror pod %s/%s", pod.metadata.namespace, pod.metadata.name)
```

Comment:
Let's use the `format` method to keep this consistent with the rest of the code.